### PR TITLE
fix: [PLG-1374]: Added referer URL

### PR DESCRIPTION
**Pull request by devinder-harness**

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]

File: src/pages/SignUp/SaasExperimentalForms/utils.ts [hunk 1]
```diff
@@ -128,4 +129,28 @@ const getModuleDetails = (moduleName = "default"): ModuleDetail => {
...
+const getURL = (str: string): URL | undefined => {
+  try {
+    return new URL(str);
+  } catch (e) {
+    return undefined;
+  }
+};
+const updateReferer = (): void => {
+  const prevRefererURL = getSavedRefererURL();
+  const harnessAppURL = "app.harness.io";
```
Comment (abhinavrastogi-harness) [inline src/pages/SignUp/SaasExperimentalForms/utils.ts:141]: why do we need to hardcode `app.harness.io`? can we use `location.hostname` instead?
how will this work in onprem env or with vanity urls?

**Review by abhinavrastogi-harness — COMMENTED** [review]
why do we have a 15k line change in cdng service?

File: src/pages/SignUp/SaasExperimentalForms/utils.ts (hunk 1 above)
Comment (devinder-harness) [reply src/pages/SignUp/SaasExperimentalForms/utils.ts:141]: this change is not required for `onprem` or `vanity` URLs. This is required only for users who signup to our site, so mostly `free` users

Comment (devinder-harness) [reply src/pages/SignUp/SaasExperimentalForms/utils.ts]: updated

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**